upup/pkg/fi/assettasks/copyfile.go

```
@@ -36,22 +36,38 @@ type CopyFile struct {
 	Name       *string
 	SourceFile *string
 	TargetFile *string
-	// @justinsb not sure I like this but the problem is knowing if the sha is a file or a string
-	// so I did a location or a sha.  We have both SHAs in remote file locations, and SHAs in strings
-	SourceShaLocation *string
-	SourceSha         *string
-	TargetSha         *string
-	Lifecycle         *fi.Lifecycle
+	SourceSha  *string
+	TargetSha  *string
+	Lifecycle  *fi.Lifecycle
 }
 
 var _ fi.CompareWithID = &CopyFile{}
 
 func (e *CopyFile) CompareWithID() *string {
+	// or should this be the SHA?
 	return e.Name
 }
 
 func (e *CopyFile) Find(c *fi.Context) (*CopyFile, error) {
-	// TODO do a head call on the file??
+	targetSHA := fi.StringValue(e.TargetFile) + ".sha1"
+	shaBytes, err := vfs.Context.ReadFile(targetSHA)
+	if err != nil {
+		return nil, nil
```

Don't just swallow the error; it should be returned or logged at _some_ level